include/libswiftnav/signal.h

```diff
@@ -61,6 +80,10 @@ static inline bool sid_is_equal(const gnss_signal_t a, const gnss_signal_t b)
   return sid_compare(a, b) == 0;
 }
 
+/** Constructs and returns a gnss_signal_t structure using given parameters.
+ */
+gnss_signal_t construct_sid(enum code code, u16 sat);
```

Comment:
Nothing significantly better for that use case. I think it's fine as long as we only use it where necessary. (For example, NOT to initialize a huge list of SIDs)